plugins/module_utils/foreman_helper.py

```diff
@@ -349,7 +349,7 @@ def ensure_entity(self, resource, desired_entity, current_entity, params=None, s
         if entity_spec is None:
             entity_spec = self.entity_spec
         else:
-            entity_spec, _ = _entity_spec_helper(entity_spec)
+            entity_spec = _entity_spec_helper(entity_spec)[0]
```

Comment:
I am not strictly happy about this one, but using `_argument_spec_ignored` seemed also weird?